### ISSUE TITLE
Can't make a valid file list

### ISSUE THREAD

**Issue by jsorge** [nicklockwood/SwiftFormat#1062]
I'm working on a script that will assemble a file list based on `git diff` output, and the call SwiftFormat with that file list. Here's a sample of the file list contents:

```
./.swift-version
./.swiftformat
./.swiftformat-version
./App.xcodeproj/project.pbxproj
./App/Application/AppDelegate.swift
./App/Backend/API/AppAPI+QuestionRatings.swift
./App/Extensions/UserDefaults+App.swift
./App/Localizable/L10n
./App/Localizable/en.lproj/Localizable.stringsdict
./App/Localizable/es.lproj/Localizable.stringsdict
./App/MainCoordinator.swift
./App/Model/Game
./App/Model/HomeScreen.swift
./App/Resources/GameListOffline/Prod/gameList.txt
./App/Resources/GameListOffline/Staging/gameList.txt
./App/ViewControllers/GameIntroductionViewController.swift
./App/ViewControllers/MenuViewController.swift
./App/ViewControllers/QuestionInputViewController.swift
./App/Views/FeatureGamePopup.swift
```
When I run the script in my repo's root, I get this error:

```error: Failed to read file list at ./vendor/SwiftFormat-Input.```

And here is the invocation of SwiftFormat

```/vendor/swiftformat --filelist "$FILE_LIST" --verbose```

Where `$FILE_LIST` is the path to the generated file that I made earlier in the script. I've tried changing the files on each line to be a full path, a relative path, and just file names. In every occasion I get the error noted above. I'm sure that I'm holding this wrong but I can't figure out how. Any ideas?

**Comment by nicklockwood** [comment]
@jsorge from the error message, it sounds like the issue is that SwiftFormat cannot find the actual `filelist` file itself at `./vendor/SwiftFormat-Input`, not that it can't find the files listed inside the file.

This probably *should* work so I'll take a look at that, but in the meantime changing the `--filelist` parameter to an absolute path (or even just removing the `./` at the beginning) ought to fix it.

**Comment by nicklockwood** [comment]
@jsorge OK, I was able to reproduce the problem, I think. Forget what I said before - the error is probably with one of the files referenced from inside your `SwiftFormat-Input` file, but SwiftFormat doesn't report which one.

Try removing all the files from the list, then put them back one at a time until the error occurs, then you'll know which one was wrong. Note: the file paths in that file should be relative to the `SwiftFormat-Input` file itself, not the current directory (but absolute paths should also work).

**Comment by jsorge** [comment]
@nicklockwood Sorry I totally forgot to update this after I figured out what was going on! I downloaded the repo and ran it locally, and discovered that the full path to the directory where the file list was being kept got prepended to each file path. I was keeping the file inside a directory nested inside of the repo so SwiftFormat kept looking for the files relative to that nested directory rather than the working directory (which was the repo root). This was the case even when I tried using absolute paths too.

**Comment by nicklockwood** [comment]
@jsorge actually, I see you have some non-swift files in your input file list, like `./App/Localizable/en.lproj/Localizable.stringsdict` - that's probably what's causing the problem. If you filter out the non-swift files from the list, that may solve it.

**Comment by jsorge** [comment]
Yeah I also added a filter in my script to filter out the non-Swift files.

**Comment by nicklockwood** [comment]
@jsorge ah OK, thanks. It's weird that would happen with absolute paths - I'll take a look into that.

**Comment by nicklockwood** [comment]
I've fixed the issues with error reporting. I'm not sure if there's another bug or not.

**Comment by nicklockwood** [comment]
@jsorge fix landed in 0.48.18
